subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -383,6 +706,28 @@ int fp_storage_ak_save(const struct fp_account_key *account_key)
 		return err;
 	}
 
+	if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(conn_ctx);
+
+		bond = fp_bond_info_by_conn_get(conn_ctx);
+		if (!bond) {
+			return -EINVAL;
+		}
+
+		__ASSERT(bond->bonded, "Account Key cannot be written by unbonded connection.");
```

Comment:
`fp_bond writing an Account Key must be already bonded`? Keep in mind that Account Key is written also by Locator Tags that do not bond at all